{{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/{{cookiecutter.django_default_app_name}}/metrics.py

```diff
@@ -36,12 +39,17 @@ def metrics_view(request):
         return ExportToDjangoView(request)
 
 
-num_tasks_in_queue = {}
-for queue in settings.CELERY_TASK_QUEUES:
-    gauge = prometheus_client.Gauge(
-        f"celery_{queue.name}_queue_len",
-        f"How many tasks are there in '{queue.name}' queue",
-    )
-    num_tasks_in_queue[queue.name] = gauge
-    gauge.set_function(partial(get_num_tasks_in_queue, queue.name))
+class CustomCeleryCollector(Collector):
+    def collect(self) -> Iterator[Metric]:
+        num_tasks_in_queue = GaugeMetricFamily(
+            "celery_queue_len",
+            "How many tasks are there in a queue",
+            labels=("queue",),
+        )
+        for queue in settings.CELERY_TASK_QUEUES:
```

Comment:
The same thing here, I don't think we need to iterate queues on each collect() invocation

Reply:
This actually seems OK to me, what's wrong? On each metrics collection we get list of queues and for each queue we ask redis queue length.

Reply:
`settings.CELERY_TASK_QUEUES` is a static value that won't be changing during the app runtime, so is there a reason to iterate and create a new metric each time? I expect prometheus to be smart enough to overwrite it, but why do it in the first place?